scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -2367,7 +2388,9 @@ class FormatOps(
         allowMain: Boolean
     )(implicit line: sourcecode.Line, style: ScalafmtConfig): Seq[Split] = {
       val multiStat = isTreeMultiStatBlock(tree)
-      val forceNL = multiStat || shouldBreakInOptionalBraces(nft)
+      val singleNoTerms = getTermSingleStat(tree).exists(_.isNot[Term])
```

Comment:
same logic as above. remove multistat, change here exists to forall. also, the check inside repeats, should we create a helper function to share with line 2098 above?

Reply:
Created a helper function.